projects/aca-content/src/lib/components/toolbar/view-node/view-node.component.ts

```diff
@@ -53,22 +54,26 @@ import { SharedLinkEntry } from '@alfresco/js-api';
 export class ViewNodeComponent {
   @Input() data: { title?: string; menuButton?: boolean; iconButton?: boolean };
 
-  constructor(private store: Store<AppStore>, private router: Router) {}
+  constructor(private store: Store<AppStore>, private router: Router, private fileAutoDownloadService: AcaFileAutoDownloadService) {}
 
   onClick() {
     this.store
       .select(getAppSelection)
       .pipe(take(1))
       .subscribe((selection) => {
-        let id: string;
-
-        if (selection.file.entry.nodeType === 'app:filelink') {
-          id = selection.file.entry.properties['cm:destination'];
+        if (this.fileAutoDownloadService.shouldFileAutoDownload(selection.file.entry?.content?.sizeInBytes)) {
+          this.fileAutoDownloadService.autoDownloadFile(selection.file);
```

Comment:
there's already store effects in this file, so better having an effect that a separate service

Reply:
The service currently performs 2 functions - Evaluate whether a file needs to downloaded or not (done via `service.shouldFileAutoDownload(fileSizeInBytes)` method), and open the prompt to download the file in case the first method returns true (done via `service.autoDownloadFile(node)` method). In case the first method returns a false value, the application should behave normally as it has been so far and execute the code that was present before. 

I understand that it is easy enough to move the second functionality inside a store effect, but how can we move the first one while making sure we have a back reference to execute the existing code in case the file should not be downloaded?